src/server-core/src/lib.rs

```diff
@@ -169,7 +171,7 @@ where
         let timedout = tokio::time::timeout(wait, async {
             while let Some(res) = set.join_next().await {
                 if let Err(e) = res {
-                    debug!(
```

Comment:
Don't we want the conn UUID logged here?

Reply:
(And then we can plug it in with the same `conn_uuid` field name, and so target it with the same tracing filter.)

Reply:
Spitballing, but I feel like you might want to consider changing `server_core::Server::handle_connection` to pass a `&mut Conn` rather than a `TcpStream` directly:

```rust
struct Conn {
    uuid: Option<Uuid>,
    tcp_stream: TcpStream,
}

And then you'd be able to log the `uuid` on connection end here if it was set by the server.

Reply:
Hmm, unfortunately, I don't think it's very straightforward to print the conn_uuid at this layer. In this function we only have access to the TCP messages, however in the adapter we need the PostgreSQL startup message to extract the UUID from the balancer. I'm not exactly sure how we'd do this correctly, but I'll think about it and happy to brainstorm if you have time today.